src/metabase/models/model_index.clj

```diff
@@ -143,7 +143,8 @@
               (doseq [deletions-part (partition-all 10000 deletions)]
                 (t2/delete! ModelIndexValue
                             :model_index_id (:id model-index)
-                            :model_pk [:in (->> deletions-part (map first))])))
+                            :model_pk [:in (->> deletions-part (map first))])
+                (search/delete! #_ModelIndexValue "indexed-entity" (->> deletions-part (map first)))))
```

Comment:
I'm not exactly sure what's the best way to approach that. I know for sure I don't want to fetch the instances beforehand, but if I supply a model here - how do I know what to delete? It's not that hard in case of `ModelIndexValue`, but is way harder in general case.